src/Engine/EmitMSIL/Replication.cs

```diff
@@ -280,7 +294,29 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
             return new List<List<ReplicationGuide>>();
         }
 
+        private static List<CLRFunctionEndPoint> GetCandidateFunctions(
+            CLRStackValue thisPtr, Dictionary<CLRFunctionEndPoint, int> candidatesWithDistances)
+        {
+            var candidateFunctions = new List<CLRFunctionEndPoint>();
+            foreach (var fep in candidatesWithDistances.Keys)
+            {
+                bool isClassInstanceMethod = fep.procedureNode != null &&
+                  fep.procedureNode.ClassID != Constants.kInvalidIndex &&//valid class
+                  !fep.procedureNode.IsConstructor && !fep.procedureNode.IsStatic;//not static and not constructor  
+
+                if (thisPtr.IsGlobal && isClassInstanceMethod)
```

Comment:
huh? Can you explain, whats a global pointer in this context?

Reply:
thisPtr.IsGlobal is my attempt to represent the global function scope (temporary solution so that we can track the old VM functionality). It is false by default for now.

It is supposed to represent design script function definitions (the global scope)
ex
CodeBlock
```
def funcA(a:int) {
return a+a;
}
```
I can remove it too and we can just keep a TODO or something ....

Reply:
I removed the flag from CLRStackValue. I left the isGLobal as hardcoded to false + TODO to figure it out later